src/Compilers/CSharp/Portable/Generated/CSharp.Generated.g4

```diff
@@ -348,7 +348,28 @@ array_rank_specifier
   ;
 
 function_pointer_type
-  : 'delegate' '*' syntax_token? '<' parameter (',' parameter)* '>'
+  : 'delegate' '*' function_pointer_calling_convention? function_pointer_parameter_list
+  ;
+
+function_pointer_calling_convention
+  : 'managed' function_pointer_unmanaged_calling_convention_syntax_list?
+  | 'unmanaged' function_pointer_unmanaged_calling_convention_syntax_list?
+  ;
+
+function_pointer_unmanaged_calling_convention_syntax_list
+  : '[' function_pointer_unmanaged_calling_convention (',' function_pointer_unmanaged_calling_convention)* ']'
+  ;
+
+function_pointer_unmanaged_calling_convention
+  : identifier_token
+  ;
+
+function_pointer_parameter_list
+  : '<' function_pointer_parameter (',' function_pointer_parameter)* '>'
+  ;
+
+function_pointer_parameter
+  : attribute_list* modifier* type
```

Comment:
i just realized something.  we don't allow identifiers for these parameters.  However, does this mean that if i have a funcptr, i can't used named-parameters with it?

Reply:
Correct. There's no place to encode parameter names in metadata.